EIPS/eip-4834.md

```diff
@@ -148,19 +146,25 @@ function resolve(string[] calldata splitName, IDomain root) public view returns
 
 Solidity Interface with NatSpec & OpenZeppelin v4 Interfaces (also available at [IDomainEnumerable.sol](../assets/eip-4834/IDomainEnumerable.sol)):
 
-```sol
+```solidity
 interface IDomainEnumerable is IDomain {
-    /// @notice     Query all subdomains. Must revert if the list of domains is unknown or infinite.
-    /// @return     The list of all subdomains.
-    function listDomains() external view returns (string[] memory);
+    /// @notice     Query all subdomains. Must revert if the number of domains is unknown or infinite.
+    /// @return     The subdomain with the given index.
+    function subdomainByIndex(uint256 index) external view returns (string memory);
+    
+    /// @notice     Get the total number of subdomains. Must revert if the number of domains is unknown or infinite.
+    /// @return     The total number of subdomains.
+    function totalSubdomains() external view returns (uint256);
```

Comment:
Since this is an actual change in the specification, I'd recommend restarting the last call period.